crates/weaver_live_check/src/advice.rs

```diff
@@ -157,23 +160,48 @@ impl Advisor for StabilityAdvisor {
 /// An advisor that checks if an attribute has the correct type
 pub struct TypeAdvisor;
 
-/// Checks if required attributes from a resolved group are present in a list of attributes
-/// Returns a list of advice for missing attributes
-fn check_required_attributes(
-    required_attributes: &[Attribute],
-    attributes: &[SampleAttribute],
+/// Checks if attributes from a resolved group are present in a list of sample attributes
+/// Returns a list of advice for the attributes based on their RequirementLevel
+fn check_attributes(
+    semconv_attributes: &[Attribute],
+    sample_attributes: &[SampleAttribute],
```

Comment:
It would be nice to have the same summary table you created in the PR description included in this comment, to map requirement levels with advice levels.